raft/raft.go

```diff
@@ -702,9 +702,9 @@ func (r *raft) becomeFollower(term uint64, lead uint64) {
 }
 
 func (r *raft) becomeCandidate() {
-	// TODO(xiangli) remove the panic when the raft implementation is stable
 	if r.state == StateLeader {
-		panic("invalid transition [leader -> candidate]")
+		r.logger.Warning("invalid transition [leader -> candidate]")
```

Comment:
In my opinion, warning is not enough when  "r.state" is wrong .
For example, "becomeCandidate"  is called by "func (r *raft) campaign(t CampaignType)",  function "campaign" will continue code even if "r.state" is wrong, it could lead to lead another panic

Reply:
If becomeCandidate() returns an error in this case, campaign() also needs to return error.

Is that the way to go ?

Thanks